pkg/cloud-provider/cloudapi/azure/azure_compute.go

```diff
@@ -348,11 +349,17 @@ func (computeCfg *computeServiceConfig) GetVpcInventory() map[string]*runtimev1a
 		return nil
 	}
 
+	vnetIDs := computeCfg.getManagedVnetIDs()
+
 	// Convert to kubernetes object and return a map indexed using VnetID.
 	vpcMap := map[string]*runtimev1alpha1.Vpc{}
 	for _, vpc := range snapshot.(*computeResourcesCacheSnapshot).vnets {
 		if strings.EqualFold(*vpc.Location, computeCfg.credentials.region) {
-			vpcObj := ComputeVpcToInternalVpcObject(&vpc, tokens[0], tokens[1], strings.ToLower(computeCfg.credentials.region))
+			managed := false
+			if _, ok := vnetIDs[strings.ToLower(*vpc.ID)]; ok {
```

Comment:
It looks like when we add vnetIDs to snapshot (in DoResourceInventory function), we do not convert vpc.ID to lowercase, hence I think we should avoid conversion to lower case while fetching from vnetIDs map here.

Reply:
we are fetching id from vm struct in doRInventory and that is in lower case.. so lower case is required here.

Reply:
Yes right, I verified it.